src/nvim/ops.c

```diff
@@ -1686,9 +1686,10 @@ int op_replace(oparg_T *oap, int c)
 
       oldp = get_cursor_line_ptr();
       oldlen = STRLEN(oldp);
-      assert(n >= 0);
-      newp = (char_u *)xmalloc(oldlen + 1 + (size_t)n);
-      memset(newp, NUL, oldlen + 1 + (size_t)n);
+      assert(oldlen <= (size_t)(INT_MAX - 1 - n));
```

Comment:
If `n` is e.g. `-3` this check will fail  (e.g. `INT_MAX -1 - (-3)` => ). I guess we need:

    assert(n < 0 || oldlen <= (size_t)(INT_MAX - 1 - n));

_Or_ let's just drop this check and say that the assert below is enough. This is getting ridiculous :>